### PR TITLE
HSEARCH-2723 Clarify the documentation about built-in bridges

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-2723

### CODE REVIEW

File: documentation/src/main/asciidoc/mapping.asciidoc [hunk 1]
```diff
@@ -1428,41 +1419,48 @@ full text representation.
...
+`short`, `Short`, `integer`, `Integer`, `long`, `Long`, `float`, `Float`, `double`, `Double`::
+Are per default indexed numerically using a link:http://en.wikipedia.org/wiki/Trie[Trie structure] which
+allows for efficient range queries and sorting, resulting in better query response times.
+You need to use a `NumericRangeQuery` to search for values. See also <<field-annotation>> and
+<<numeric-field-annotation>>.
+
+[CAUTION]
+====
+Prior to Search 5, numeric field encoding was only chosen if explicitly requested via `@NumericField`.
+As of Search 5 this encoding is automatically chosen for the numeric types mentioned above.
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapping.asciidoc]: It was preexisting, but I think we should use `Hibernate Search` not `Search`.

Maybe a comma after `As of (Hibernate) Search 5` would be in order.

File: documentation/src/main/asciidoc/mapping.asciidoc [hunk 2]
```diff
@@ -1428,41 +1419,48 @@ full text representation.
...
+
+To avoid numeric encoding you can explicitly specify a non numeric field bridge via `@Field.bridge` or `@FieldBridge`.
+The package `org.hibernate.search.bridge.builtin` contains a set of bridges which encode numbers as strings,
+for example `org.hibernate.search.bridge.builtin.IntegerBridge`.
+====
 
+`byte`, `Byte`, `short`, `Short`::
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapping.asciidoc]: Looks like `short` and `Short` were already mentioned above in the numeric fields?

File: documentation/src/main/asciidoc/mapping.asciidoc [hunk 3]
```diff
@@ -1428,41 +1419,48 @@ full text representation.
...
 with a potential loss in precision can be seen in <<example-custom-numericfieldbridge>>.
 
 `java.util.Date`, `java.util.Calendar`:: Dates are indexed as `long` value representing the number
 of milliseconds since _January 1, 1970, 00:00:00 GMT_. You shouldn't really bother with the
 internal format. It is important, however, to query a numerically indexed date via a `NumericRangeQuery`.
 +
-Usually, storing the date up to the millisecond is not necessary. `@DateBridge` defines the
-appropriate resolution you are willing to store in the index.
+Usually, storing the date up to the millisecond is not necessary.
+`@DateBridge` and `@CalendarBridge` define the appropriate resolution you are willing to store in the index.
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapping.asciidoc]: Maybe `allows to define`?

File: documentation/src/main/asciidoc/mapping.asciidoc [hunk 4]
```diff
@@ -1475,22 +1473,36 @@ public class Meeting {
...
-
++
 [IMPORTANT]
 ====
 A Date whose resolution is lower than `MILLISECOND` cannot be a `@DocumentId`
 ====
-
++
+You can also choose to encode the date as string using the `encoding=EncodingType.STRING` of
+`DateBridge`. In this case the dates are stored in the format _yyyyMMddHHmmssSSS_ (using GMT time).
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapping.asciidoc]: `**@**DateBridge` to be consistent with above. And maybe mention `@CalendarBridge` too.

File: documentation/src/main/asciidoc/mapping.asciidoc [hunk 5]
```diff
@@ -1475,22 +1473,36 @@ public class Meeting {
...
 ----
 ====
-
-You can also choose to encode the date as string using the `encoding=EncodingType.STRING` of
-`DateBridge`. In this case the dates are stored in the format _yyyyMMddHHmmssSSS_ (using GMT time).
-
++
 [IMPORTANT]
 ====
 A Date whose resolution is lower than `MILLISECOND` cannot be a `@DocumentId`
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapping.asciidoc]: This looks pretty arbitrary and weird. Do we really have a safeguard for this?

File: documentation/src/main/asciidoc/mapping.asciidoc (hunk 1 above)
Comment (yrodiere) [inline documentation/src/main/asciidoc/mapping.asciidoc]: Done.

File: documentation/src/main/asciidoc/mapping.asciidoc (hunk 2 above)
Comment (yrodiere) [inline documentation/src/main/asciidoc/mapping.asciidoc]: You're right: it was a pre-existing mistake. I'll remove those types from the list above.

File: documentation/src/main/asciidoc/mapping.asciidoc (hunk 4 above)
Comment (yrodiere) [inline documentation/src/main/asciidoc/mapping.asciidoc]: Done.

File: documentation/src/main/asciidoc/mapping.asciidoc (hunk 3 above)
Comment (yrodiere) [inline documentation/src/main/asciidoc/mapping.asciidoc]: Done.

File: documentation/src/main/asciidoc/mapping.asciidoc (hunk 5 above)
Comment (yrodiere) [inline documentation/src/main/asciidoc/mapping.asciidoc]: I can see why we would enforce this: truncating IDs in any way is bad idea.
But no, this doesn't seem to be enforced anywhere: the only places we call `org.hibernate.search.annotations.DateBridge.resolution()` from are the date/calendar bridge providers, and we don't  check anything about the ID.

Maybe I should simply remove this warning... ? After all, you can also shoot yourself in the foot by assigning any custom field bridge to the document ID.

Comment (gsmet) [inline documentation/src/main/asciidoc/mapping.asciidoc]: Yeah, if we don't have specific code to check for this, let's remove this.

**Comment by gsmet** [comment]
Rebased and merged. With one additional commit s/integer/int/ I missed on the first review.

**Pull request closed without merging**